### PR TITLE
Fix redemption unit calculation for basic issuance and handling of custom errors from 0x router

### PR DESCRIPTION
**Pull request by ckoopmann**

### CODE REVIEW

File: contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol [hunk 1]
```diff
@@ -536,11 +546,12 @@ contract ExchangeIssuanceZeroEx is Ownable, ReentrancyGuard {
      * @param _setToken          Set token to issue
      * @param _amountSetToken    Amount of set token to issue
      */
-    function _getRequiredIssuanceComponents(address _issuanceModule, ISetToken _setToken, uint256 _amountSetToken) internal  returns(address[] memory components, uint256[] memory positions) {
+    // I can't make this a view since the methods on the issuance module interface are not marked as view
+    //  TODO: Review if the interfaces are correct or should be changed
+    function getRequiredIssuanceComponents(address _issuanceModule, ISetToken _setToken, uint256 _amountSetToken) public returns(address[] memory components, uint256[] memory positions) {
```
Comment (ncitron) [inline contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol:551]: The interface is incorrect. Feel free to make it view.

File: contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol [hunk 2]
```diff
@@ -536,11 +546,12 @@ contract ExchangeIssuanceZeroEx is Ownable, ReentrancyGuard {
...
         if(allowedIssuanceModules[_issuanceModule].isDebtIssuanceModule) {
             // TODO: Check if possible to ignore third return parameter and avoid declaring an unused variable
```
Comment (ncitron) [inline contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol:553]: Think this comment is old now?

**Review by ncitron — COMMENTED** [review]
LGTM! Just a couple nits. Once these are fixed you can open a PR against the main repo. Then I'll do a more holistic review where I can look at everything and focus on security. I'm not sure if Set also wants to review this since they might be integrating it into their frontend but if they are they can probably begin their review in parallel.

File: contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol (hunk 2 above)
Comment (ckoopmann) [reply contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol:553]: Yes it is. will delete

File: contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol (hunk 1 above)
Comment (ckoopmann) [reply contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol:551]: Will do, thanks

Comment (ckoopmann) [reply contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol]: Adjusted to view

**Pull request merged**